File: server/Gate/Connector/ServerConnector.cs
//////////////////////////////////////////////////////////////////////////
// 
// 文件：server/Gate/Connector/ServerConnector.cs
// 作者：Xoen Xie
// 时间：2023/06/19
// 描述：消息处理
// 说明：
//
//////////////////////////////////////////////////////////////////////////
#pragma warning disable CS8600, CS8602
using XSF;

public class ServerConnector : NetConnector
{
    private ConnectorManager m_Owner;

    public ServerConnector(ConnectorManager owner)
    {
        m_Owner = owner;
    }

    public override void SendHandshake()
    {
        var message = XSFCore.GetMessage((ushort)XsfPbid.SMSGID.GtGtAHandshake) as XsfMsg.MSG_Gt_GtA_Handshake;
        message.mPB.ServerId = XSFCore.Server.ID;
        SendMessage(message);
    }

    public override void SendHeartbeat()
    {
        var message = XSFCore.GetMessage((ushort)XsfPbid.SMSGID.GtGtAHeartbeat);
        SendMessage(message);
    }

    public override void OnNetError()
    {
        m_Owner.DeleteConnector(m_nID);
    }
}

File: server-cpp/source/gate/server/src/connector/ServerConnector.cpp
//////////////////////////////////////////////////////////////////////////
//
// 文件：source/gate/server/src/connector/ServerConnector.cpp
// 作者：Xoen Xie
// 时间：2020/08/06
// 描述：连接器
// 说明：
//
//////////////////////////////////////////////////////////////////////////
#include "ServerConnector.h"
#include "DMessageInc.h"
#include "ConnectorManager.h"

void ServerConnector::SendHandshake(void)
{
    XSF_CAST(pMessage, XSFCore::GetMessage(xsf_pbid::Gt_GtA_Handshake), xsf_msg::MSG_Gt_GtA_Handshake);
    pMessage->mPB.set_server_id(XSFCore::GetServer()->GetSID()->ID);
    SendMessage(pMessage);
}

void ServerConnector::SendHeartbeat(void)
{
    SendMessage(XSFCore::GetMessage(xsf_pbid::Gt_GtA_Heartbeat));
}

void ServerConnector::OnNetError(void)
{
    ConnectorManager::Instance()->DeleteConnector(m_RemoteID.ID);
}